### PR TITLE
Select last or specific boundary set for process configuration

### PR DESCRIPTION
**Pull request by FranckLecuyer**
PR that must be merged before (migration merge orchestrator to Postgres) : 
https://github.com/gridsuite/merge-orchestrator-server/pull/50
https://github.com/gridsuite/deployment/pull/145

All following PR must be merged together : 
https://github.com/gridsuite/cgmes-boundary-server/pull/15
https://github.com/gridsuite/cgmes-boundary-import-job/pull/3
https://github.com/gridsuite/merge-orchestrator-server/pull/51
https://github.com/gridsuite/merge-notification-server/pull/16
https://github.com/gridsuite/gridmerge-app/pull/67

### CODE REVIEW

File: src/components/process.js [hunk 1]
```diff
@@ -85,19 +85,25 @@ const Process = (props) => {
...
-            const state = store.getState();
-            const processDate = state.processes[props.index].date;
-
-            // if same day as selected, reload merges from server
-            if (removeTime(mergeDate).getTime() === processDate.getTime()) {
-                loadMerges(processDate);
+            if (headers['date']) {
+                const mergeDate = new Date(headers['date']);
+
+                // we need to directly access the store to get current process date as the wensocket message handler cannot
```
Comment (SlimaneAmar) [inline src/components/process.js]: websocket typo ?

File: src/components/processes-configuration-dialog.js [hunk 2]
```diff
@@ -171,6 +172,133 @@ const ProcessTsos = ({
     );
 };
 
+const ProcessBoundarySet = ({
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js:175]: Is really a good pratice to have several composants in the same file ?

File: src/components/processes-configuration-dialog.js [hunk 3]
```diff
@@ -220,6 +350,33 @@ const ProcessesContainer = ({ handleProcessesChanged, currentProcess }) => {
         handleProcessesChanged(currentProcessesCopy);
     }
 
+    function handleProcessUseBoundarySetChanged(e, index) {
+        const currentProcessesCopy = [...currentProcess];
+        currentProcessesCopy[index] = {
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js:347]: currentProcessesCopy[index]. useLastBoundarySet = (e.target.value === 'true') ?

File: src/components/processes-configuration-dialog.js [hunk 4]
```diff
@@ -171,6 +172,133 @@ const ProcessTsos = ({
...
+
+    return (
+        <>
+            <RadioGroup
+                row
+                aria-label="useBoundarySet"
+                name="useBoundarySet"
+                value={
+                    process.useLastBoundarySet !== null
+                        ? process.useLastBoundarySet + ''
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js]: toString ?

File: src/components/processes-configuration-dialog.js [hunk 5]
```diff
@@ -171,6 +172,133 @@ const ProcessTsos = ({
...
+                    })}
+                />
+            </RadioGroup>
+            <Grid container spacing={2}>
+                <Grid item xs={12} sm={10}>
+                    <Autocomplete
+                        disabled={disabledBoundaries}
+                        id="select_eq_boundary"
+                        value={
+                            process.eqBoundary !== null
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js]: ?

File: src/components/processes-configuration-dialog.js [hunk 6]
```diff
@@ -220,6 +350,33 @@ const ProcessesContainer = ({ handleProcessesChanged, currentProcess }) => {
...
+        currentProcessesCopy[index] = {
+            ...currentProcessesCopy[index],
+            ...{ useLastBoundarySet: e.target.value === 'true' },
+        };
+        handleProcessesChanged(currentProcessesCopy);
+    }
+
+    function handleProcessEqBoundarySetChanged(value, index) {
+        const currentProcessesCopy = [...currentProcess];
+        currentProcessesCopy[index] = {
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js:356]: currentProcessesCopy[index]. eqBoundary = value ?

File: src/components/processes-configuration-dialog.js [hunk 7]
```diff
@@ -220,6 +350,33 @@ const ProcessesContainer = ({ handleProcessesChanged, currentProcess }) => {
...
+            ...currentProcessesCopy[index],
+            ...{ eqBoundary: value },
+        };
+        handleProcessesChanged(currentProcessesCopy);
+    }
+
+    function handleProcessTpBoundarySetChanged(value, index) {
+        const currentProcessesCopy = [...currentProcess];
+        currentProcessesCopy[index] = {
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js:365]: currentProcessesCopy[index]. tpBoundary = value ?

File: src/components/processes-configuration-dialog.js [hunk 8]
```diff
@@ -438,10 +614,26 @@ const ProcessesConfigurationDialog = ({ open, onClose, matchProcess }) => {
...
         };
     };
 
+    const isBoundaryIdentical = (initialBoundary, currentBoundary) => {
+        return (
+            (initialBoundary == null && currentBoundary == null) ||
+            (initialBoundary !== null &&
+                currentBoundary !== null &&
+                initialBoundary.filename === currentBoundary.filename)
+        );
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js]: let initialFilename = initialBoundary ? initialBoundary.filename : null;
let currentFilename = currentBoundary ? currentBoundary.filename : null;
return initialFilename === currentFilename;

File: src/components/processes-configuration-dialog.js [hunk 9]
```diff
@@ -481,7 +689,7 @@ const ProcessesConfigurationDialog = ({ open, onClose, matchProcess }) => {
                 (element) => element.processUuid === processes[i].processUuid
             );
 
-            if (typeof initialProcess === 'undefined') {
+            if (initialProcess == null) {
```
Comment (SlimaneAmar) [inline src/components/processes-configuration-dialog.js]: if (initialProcess) ?

File: src/components/process.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/process.js]: Corrected

File: src/components/processes-configuration-dialog.js (hunk 4 above)
Comment (FranckLecuyer) [reply src/components/processes-configuration-dialog.js]: Done

File: src/components/processes-configuration-dialog.js (hunk 5 above)
Comment (FranckLecuyer) [reply src/components/processes-configuration-dialog.js]: Done

File: src/components/processes-configuration-dialog.js (hunk 8 above)
Comment (FranckLecuyer) [reply src/components/processes-configuration-dialog.js]: Done

File: src/components/processes-configuration-dialog.js (hunk 9 above)
Comment (FranckLecuyer) [reply src/components/processes-configuration-dialog.js]: Done

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**